src/colmap/estimators/generalized_pose.cc

```diff
@@ -0,0 +1,306 @@
+// Copyright (c) 2023, ETH Zurich and UNC Chapel Hill.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+//     * Neither the name of ETH Zurich and UNC Chapel Hill nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+// Author: Johannes L. Schoenberger (jsch-at-demuc-dot-de)
+
+#include "colmap/estimators/generalized_pose.h"
+
+#include "colmap/estimators/bundle_adjustment.h"
+#include "colmap/estimators/cost_functions.h"
+#include "colmap/estimators/generalized_absolute_pose.h"
+#include "colmap/estimators/pose.h"
+#include "colmap/geometry/rigid3.h"
+#include "colmap/math/matrix.h"
+#include "colmap/optim/ransac.h"
+#include "colmap/optim/support_measurement.h"
+#include "colmap/scene/camera.h"
+#include "colmap/sensor/models.h"
+#include "colmap/util/misc.h"
+
+#include <Eigen/Core>
+#include <ceres/ceres.h>
+
+namespace colmap {
+namespace {
+
+bool LowerVector3d(const Eigen::Vector3d& v1, const Eigen::Vector3d& v2) {
```

Comment:
`(v1 < v2).any()` would be the same?

Reply:
No, here we explicitly order along the x axis first, then y, then z.